language-design.md

```diff
@@ -38,6 +147,11 @@ How do the themes of _Growing a Language_ relate to the "sound lab" we did this
 
 **Response**
 
+In the Sound lab, we brainstormed ways to modify existing files to make it easier to chain together function calls, for instance, to change the volume and reverse the sound file. Common ideas were to rename function names for clarity, and to eliminate the redundant arguments. We grappled with the benefits and drawbacks of a functional style method-chaining versus the imperative method-chaining. 
+
+In hindsight, this activity was reminiscent of Steele’s “bazaar” style of development. We were given the results of a CS5 problem set to start with; it seemed like a group of programmers had created a DSL to interact with sounds and we were essentially iterating on top to make it more intuitive to use. We had full control of how we wanted it to develop and could enjoy the fruits of our hard work if we had decided to follow through with some of our ideas.  
+
+On the other hand, because we were to first explore on our own we all had different ideas of how to approach the problem. In this sense, we got a taste of what it would be like to be the solo designer implementing our own “master plan”. Once we came back to a group discussion, it was interesting to see that many of us had similar ideas, but even more intriguing to see where we diverged. It was quite clear from this exercise that the bazaar approach is in general superior to the cathedral approach. 
```

Comment:
I hadn't thought of us creating a "master plan", but that is probably true. I'd like to say that our master plan can also be influenced by user complaints. Although the complaints (i.e. why can't we reuse 'out.wav') came after we planned our improvements, if we were to continue working on the sound lab, we would try to improve the program so that people would stop complaining. By then, it's not just one designer improving the sound lab; it's also the users who contribute to further development.